core/src/main/kotlin/Model/DocumentationReference.kt

```diff
@@ -41,7 +41,14 @@ class NodeReferenceGraph() {
     val references = arrayListOf<PendingDocumentationReference>()
 
     fun register(signature: String, node: DocumentationNode) {
-        nodeMap.put(signature, node)
+        val javaSignature = signature.replace("kotlin.Any", "java.lang.Object").
```

Comment:
This shouldn't be done on `DocumentationReference` side, it should be changed on somehow in JavaDocParser, to make it emit proper signatures for both AsJava and AsKotlin mode.